custom_components/sems/sensor.py

```diff
@@ -117,453 +163,351 @@ async def async_update_data():
+                    f"Inverter {inverter['name']} PV String {idx} Voltage",
+                    path_to_inverter + [f"vpv{idx}"],
+                    Decimal,
+                    SensorDeviceClass.VOLTAGE,
+                    ELECTRIC_POTENTIAL_VOLT,
+                    SensorStateClass.MEASUREMENT,
+                ) for idx in range(1, 5)]
+        inverter_strings += [
+                Sensor(
+                    coordinator,
+                    device_info,
+                    f"{serial_number}-vac{idx}",
+                    f"Inverter {inverter['name']} Grid {idx} AC Voltage",
+                    path_to_inverter + [f"vac{idx}"],
+                    Decimal,
+                    SensorDeviceClass.VOLTAGE,
+                    ELECTRIC_POTENTIAL_VOLT,
+                    SensorStateClass.MEASUREMENT,
+                    AC_EMPTY,
+                ) for idx in range(1, 4) ]
+        grids = [
+                Sensor(
+                    coordinator,
+                    device_info,
+                    f"{serial_number}-iac{idx}",
+                    f"Inverter {inverter['name']} Grid {idx} AC Current",
+                    path_to_inverter + [f"iac{idx}"],
+                    Decimal,
+                    SensorDeviceClass.CURRENT,
+                    ELECTRIC_CURRENT_AMPERE,
+                    SensorStateClass.MEASUREMENT,
+                    AC_CURRENT_EMPTY,
+                ) for idx in range(1, 4) ]
+        grids += [
+                Sensor(
+                    coordinator,
+                    device_info,
+                    f"{serial_number}-fac{idx}",
+                    f"Inverter {inverter['name']} Grid {idx} AC Frequency",
+                    path_to_inverter + [f"fac{idx}"],
+                    Decimal,
+                    SensorDeviceClass.FREQUENCY,
+                    FREQUENCY_HERTZ,
+                    SensorStateClass.MEASUREMENT,
+                    AC_FEQ_EMPTY,
+                ) for idx in range(1, 4)]
+        batteries = [
+                Sensor(
+                    coordinator,
+                    device_info,
+                    f"{serial_number}-vbattery1",
+                    f"Inverter {inverter['name']} Battery Voltage",
+                    path_to_inverter + ["vbattery1"],
+                    Decimal,
+                    SensorDeviceClass.VOLTAGE,
+                    ELECTRIC_POTENTIAL_VOLT,
+                    SensorStateClass.MEASUREMENT,
+                ),
+                Sensor(
+                    coordinator,
+                    device_info,
+                    f"{serial_number}-ibattery1",
+                    f"Inverter {inverter['name']} Battery Current",
+                    path_to_inverter + ["ibattery1"],
+                    Decimal,
+                    SensorDeviceClass.CURRENT,
+                    ELECTRIC_CURRENT_AMPERE,
+                    SensorStateClass.MEASUREMENT,
+                ),
+            ]
```

Comment:
@f00f 
> When configuring the Energy dashboard, I don't see sensors for battery in the dropdown list. Is the sensor simply missing from this PR? Does the api not provide the data? Can I build a custom sensor from available data?

I don't have a battery so I am making some assumptions about what these values will look like, if you could provide the API response JSON you are getting from the SEMS portal we can improve this. *Make sure you don't include any personal info* the start of the JSON normally contains your name ect.